pkg/cvo/status.go

```diff
@@ -126,8 +126,6 @@ func mergeOperatorHistory(config *configv1.ClusterVersion, desired configv1.Upda
 
 	// TODO: prune Z versions over transitions to Y versions, keep initial installed version
 	pruneStatusHistory(config, 50)
-
-	config.Status.Desired = desired
```

Comment:
I still not clear why this does not work.

Reply:
`syncStatus` is feeding `mergeOperatorHistory` `status.Actual`, which diverges from `optr.currentVersion()` when an update has been requested and we're working through preconditions.  With this commit, we're always setting `Desired` to `currentVersion()`.

Reply:
`syncStatus` is feeding `mergeOperatorHistory` `status.Actual`, which diverges from `optr.currentVersion()` when an update has been requested and we're working through preconditions.  With this commit, we're always setting `Desired` to `currentVersion()`.